### ISSUE TITLE
HorizontalPager deprecated in 1.5.0

### ISSUE THREAD

**Issue by Knoxvillekm** [arkivanov/Decompose#444]
We need to use the overload without pageCount. Deprecation level = DeprecationLevel.ERROR.

**Comment by arkivanov** [comment]
Thanks. I believe this is being addressed in #443 by @malliaridis. No deprecation is required, since the API is experimental.

**Comment by arkivanov** [comment]
Will be fixed in the next 2.1 release.

**Comment by Knoxvillekm** [comment]
Why did you change the parameter
`pages: ChildPages<*, T>,`
to
`pages: State<ChildPages<*, T>>,`

In example we use
```
val childStack by component.childStack.subscribeAsState()
Children(
    stack = childStack,
    ...
```
and want use same for pages.

**Comment by arkivanov** [comment]
@Knoxvillekm Please see the [discussion](https://github.com/arkivanov/Decompose/pull/443#discussion_r1292536081) in the PR. 

In your case you should be able to write:

```kotlin
Children(
    stack = component.childStack,
    ...
```

**Comment by arkivanov** [comment]
According to https://issuetracker.google.com/issues/295653367#comment7, the old way should still work. I will double check and revert that particular change if possible.

**Comment by arkivanov** [comment]
OK, after further investigation it appears that for now we need `pages: State<ChildPages<*, T>>` instead of just `pages: ChildPages<*, T>`. Otherwise there is `ArrayIndexOutOfBoundsException` on state change, described in this bug:  https://issuetracker.google.com/issues/295653367. As per [comment #5](https://issuetracker.google.com/issues/295653367#comment5), the crash is fixed in Compose version `1.6.0-alpha03`, so we should start using `pages: ChildPages<*, T>` again once Compose is updated to the stable version `1.6.0`. Look out of the new issue discovered in Compose version `1.6.0-alpha03`, described in the [comment #8](https://issuetracker.google.com/issues/295653367#comment8).

**Comment by arkivanov** [comment]
Filed #461 for this.
